smithy-linters/src/main/java/software/amazon/smithy/linters/MissingSensitiveTraitValidator.java

```diff
@@ -0,0 +1,233 @@
+                    .map(phrase -> phrase.toLowerCase(Locale.US))
+                    .collect(Collectors.toSet()));
+            wordsInit.addAll(config.getWords()
+                    .stream()
+                    .map(word -> word.toLowerCase(Locale.US))
+                    .collect(Collectors.toSet()));
+            sensitivePhrases = Collections.unmodifiableSet(phrasesInit);
+            sensitiveWords = Collections.unmodifiableSet(wordsInit);
+        } else {
+            if (config.getPhrases().isEmpty() && config.getWords().isEmpty()) {
+                //This configuration combination makes the validator a no-op.
+                throw new IllegalArgumentException("Cannot set 'excludeDefaults' to true and leave "
+                                                 + "both 'phrases' and 'words' unspecified.");
+            }
+            sensitivePhrases = Collections.unmodifiableSet(new HashSet<>(
+                    config.getPhrases()
+                            .stream()
+                            .map(phrase -> phrase.toLowerCase(Locale.US))
+                            .collect(Collectors.toSet()))
+            );
+            sensitiveWords = Collections.unmodifiableSet(new HashSet<>(
+                    config.getWords()
+                            .stream()
+                            .map(word -> word.toLowerCase(Locale.US))
+                            .collect(Collectors.toSet())));
+        }
+    }
+
+    /**
+     * Finds shapes without the sensitive trait that possibly contain sensitive data,
+     * based on the shape/member name and the list of key words and phrases.
+     *
+     * @param model Model to validate.
+     * @return list of violation events
+     */
+    @Override
+    public List<ValidationEvent> validate(Model model) {
+        List<ValidationEvent> validationEvents = new ArrayList<>();
+        validationEvents.addAll(scanShapeNames(model));
+        validationEvents.addAll(scanMemberNames(model));
+        return validationEvents;
+    }
+
+    private List<ValidationEvent> scanShapeNames(Model model) {
+        return model.shapes()
+                .filter(shape -> !shape.isMemberShape()
+                        && !shape.isOperationShape()
+                        && !shape.isServiceShape()
+                        && !shape.isResourceShape())
+                .filter(shape -> !shape.hasTrait(SensitiveTrait.ID))
+                .map(shape -> detectSensitiveWord(shape.toShapeId().getName(), shape).map(Optional::of)
+                        .orElse(detectSensitivePhrase(shape.toShapeId().getName(), shape)))
+                .flatMap(o -> o.map(Stream::of).orElseGet(Stream::empty))
+//                Once we finally upgrade from Java 8, above line can be simplified to
+//                .flatMap(Optional::stream)
+                .collect(Collectors.toList());
+    }
+
+    private List<ValidationEvent> scanMemberNames(Model model) {
+        return model.shapes()
```

Comment:
This and the other method are pretty inefficient. You could filter up front all shapes marked with the sensitive trait, put that in a set, then consult that when iterating over shapes. In this case, you'd only want to iterate over members too.

```
// This method is cached and might have already have been called.
Set<Shape> sensitiveShapes = model.getShapesWithTrait(SensitiveTrait.class);
```